website/docs/r/branch_protection.html.markdown

```diff
@@ -29,8 +45,16 @@ The following arguments are supported:
 
 * `branch` - (Required) Name of the branch.
 
-* `push_access_level` - (Required) One of five levels of access to the project.
+* `push_access_level` - (Required if not setting allowed_to_push) One of five levels of access to the project.
+
+* `merge_access_level` - (Required if not setting allowed_to_merge) One of five levels of access to the project.
+
+* `unprotect_access_level` - (Optional; conflicts with allowed_to_unprotect) One of five levels of access to the project.  Defaults to Maintainer.
+
+* `allowed_to_push` - (Required if not setting push_access_level) GitLab EE Only - Map of user ids, group ids, and access levels to grant access.  At least one of user_id, group_id, or access_level must be defined.
+
+* `allowed_to_merge` - (Required if not setting merge_access_level) GitLab EE Only - Map of user ids, group ids, and access levels to grant access.  At least one of user_id, group_id, or access_level must be defined.
 
-* `merge_access_level` - (Required) One of five levels of access to the project.
+* `allowed_to_unprotect` - (Optional; conflicts with unprotect_access_level) GitLab EE Only - Map of user ids, group ids, and access levels to grant access.  At least one of user_id, group_id, or access_level must be defined.
 
-* `code_owner_approval_required` (Optional) Bool, defaults to false. Can be set to true to require code owner approval before merging.
+* `code_owner_approval_required` - (Optional) GitLab EE Only - True or false.  Defaults to false.
```

Comment:
Can you go back to the old description, and maybe add `(GitLab EE Only)` at the end?

Reply:
Will do.

Reply:
Done.